rpcserver.go

```diff
@@ -4168,11 +4168,9 @@ func createRPCOpenChannel(r *rpcServer, dbChannel *channeldb.OpenChannel,
 	// Look up our channel peer's node alias if the caller requests it.
 	if peerAliasLookup {
 		peerAlias, err := r.server.graphDB.LookupAlias(nodePub)
-		if err != nil {
-			return nil, fmt.Errorf("unable to lookup peer "+
-				"alias: %w", err)
+		if err == nil {
+			channel.PeerAlias = peerAlias
```

Comment:
I think with this, it is not possible to distinguish between a present but empty alias and the error case.

Reply:
True. Do you suggest we add the error to the alias field then? Something like `<alias lookup failed: %v>`? That would kind of solve both issues, though might look bad/alarming in a UI (I can see the issues being opened about it already)...

Reply:
I think a bool field `AliasPresent`

Reply:
I like that idea, although it should probably named something like `PeerAliasPresent` to distinguish it from the scid aliases?

Reply:
If that's the final verdict then I will submit that in a bit.

Reply:
@joostjager so you'd set the `PeerAliasPresent` to `true` if there was an error fetching the alias? Even though we don't know at that point if the peer has an alias, because the lookup failed? Or would `true` indicate it was loaded without issue but might be empty? Maybe `PeerAliasFetched` would be a more appropriate name.

Reply:
I think `true` when the alias is properly set (potentially set to empty if that's what the node advertises) is the most intuitive option.

Not sure about fetching, because isn't it that the fetching is not the problem? Maybe I am wrong, but I thought that no alias means that the node didn't send out a node announcement or that we haven't received it yet?

Reply:
Okay, I guess naming is hard :) I would've interpreted the "present" as "has the peer node operator decided to give the node an alias or not". How about `PeerAliasLookupFailed` if there was an error?

Reply:
I just ran into this on regtest as well. The error is `alias for node not found`, which probably means there was no node announcement received yet (the peer definitely has an alias set). So "lookup" (or "fetch") would refer to the local graph DB.

Reply:
I think we're over complicating this... Is the `ListChannels` or `ForwardingEvents` really the right RPC to bubble up graph related errors? You always have the peer public key and a UI would very likely link that to the peer's node info page, which IMO is the right place to surface such an error. And if you're using the RPC, I don't think you're that interested in the alias. So why not just return `<n/a>` if there's an error?

Reply:
Are you saying that #7322 is not that useful? I agree to that. But in any case, I wouldn't go for returning a magic string like `<n/a>`. This is painful enough in other areas already.

Reply:
No, that's not what I'm saying. I'm saying that a mere graph sync issue (which might happen outside of regtest, on startup) shouldn't break my channel list output. So I'd rather see an empty alias (or some kind of placeholder) and wonder why it doesn't show than have an unexpected error returned.

Anyway, I think we're splitting hairs here. Possibly the issue is that in `ListChannels` we're using `LookupAlias` instead of `FetchLightningNode`, which seems to return an error more often.

Reply:
Yes, definitely. No rpc error if lnd just doesn't know what the alias is (yet).

Reply:
> And if you're using the RPC, I don't think you're that interested in the alias. So why not just return <n/a> if there's an error?

I think that's the opposite, at least from #6994. Also because if you need to explicitly enable it by passing `PeerAliasLookup` in the request.

> Something like <alias lookup failed: %v>?

I can't think of a better design than this one.

Reply:
> I think we're over complicating this... Is the `ListChannels` or `ForwardingEvents` really the right RPC to bubble up graph related errors?... So why not just return `<n/a>` if there's an error?

I like this approach too. It gives immediate feedback to the caller without interrupting the flow and spamming the logs.

>> Something like <alias lookup failed: %v>?

> I can't think of a better design than this one.

If no objection I'd go for this.